braindecode/preprocessing/preprocess.py

```diff
@@ -274,23 +275,53 @@ def zscore(data):
     return zscored
 
 
+def robust_scale(data):
+    """Robust scale continuous or windowed data in-place.
+
+    Substract median and divide by interquartile range.
+
+    Parameters
+    ----------
+    data: np.ndarray
+        Continuous or windowed data of shape (n_channels, n_times) or
+        (n_windows, n_channels, n_times).
+
+    Returns
+    -------
+    np.ndarray :
+        Normalized continuous or windowed data of shape (n_channels, n_times)
+        or (n_windows, n_channels, n_times).
+
+    .. note::
+        If this function is supposed to preprocess continuous data, it should be
+        given to raw.apply_function().
+    """
+    scaled = data - np.median(data, keepdims=True, axis=-1)
+    q75, q25 = np.percentile(scaled, [75, 25], keepdims=True, axis=-1)
+    scaled = scaled / (q75 - q25)
+    if hasattr(data, '_data'):
+        data._data = scaled
```

Comment:
I don’t like that. Docsting says data is ndarray that will be modified inplace but inplace occurs only for Evoked and Epochs (not explained from docstring) and assume hack a private attribute only present if preload = True. You should be using apply_function here.

Reply:
It was an artifact of the other preprocessors - I agree with you. I removed them with the last commit.